tests/src/test/scala/org/apache/openwhisk/core/entity/test/WhiskEntityTests.scala

```diff
@@ -162,4 +147,78 @@ class WhiskEntityTests extends FlatSpec with ExecHelpers with Matchers {
     val trigger = WhiskTrigger(namespace, name)
     assertType(trigger, "trigger")
   }
+
+  behavior of "WhiskDocumentReader"
+
+  it should "check entityType when deserialize" in {
+    def assertType(d: WhiskEntity, entityType: String) = {
+      d.toDocumentRecord.fields("entityType") shouldBe JsString(entityType)
+    }
+
+    val json = """{
+                 |	"name": "action_test",
+                 |	"publish": false,
+                 |	"annotations": [],
+                 |	"version": "0.0.2",
+                 |	"entityType": "action",
+                 |	"exec": {
+                 |		"kind": "nodejs:10",
+                 |		"code": "foo",
+                 |		"binary": false
+                 |	},
+                 |	"parameters": [],
+                 |	"limits": {
+                 |		"timeout": 60000,
+                 |		"memory": 256
+                 |	},
+                 |	"namespace": "namespace"
+                 |}""".stripMargin.parseJson
+
+    try {
+      val action = WhiskDocumentReader.read(manifest[WhiskAction], json)
+      assertType(action.asInstanceOf[WhiskEntity], "action")
+    } catch {
+      case _: DocumentTypeMismatchException =>
+        assert(false)
+    }
+
+    try {
```

Comment:
We can simplify this by using [scalatest assertThrows](http://www.scalatest.org/user_guide/using_assertions#interceptedExceptions)